### PR TITLE
fix #1898

### LINKED ISSUE #1898: Recent title reading changes break normal builds
```rest
.. title: Creating a Site (Not a Blog) with Nikola
.. slug: creating-a-site-not-a-blog-with-nikola
.. date: 2015-01-10 10:00:00 UTC
.. tags: nikola, python
.. link:
.. description:

Creating a Site (Not a Blog) with Nikola
========================================
```

```
Scanning posts......[2015-07-20T14:50:30Z] ERROR: Nikola: Two posts are trying to generate index.html: stories/creating-a-site.txt and stories/index.html
[2015-07-20T14:50:30Z] ERROR: Nikola: Two posts are trying to generate index.html: stories/creating-a-site.txt and stories/index.html
done!
```

git bisect: 7491fe1e1d23fbe952a0456ff9f88b738f951c34

### PR DESCRIPTION
**Pull request by ralsina**
Rewrite crazy metadata extraction thingie.

### CODE REVIEW

File: nikola/post.py [hunk 1]
```diff
@@ -857,40 +857,38 @@ def _get_metadata_from_file(meta_data):
...
             break
-        if 'title' not in meta:
-            match = re_meta(line, 'title')
-            if match[0]:
-                meta['title'] = match[1]
-        if 'title' not in meta:
+        match = re_meta(line)
+        if match[0]:
+            meta[match[0]] = match[1]
+            seen_meta_flag = True
```
Comment (Kwpolska) [inline nikola/post.py]: unused

File: nikola/post.py [hunk 2]
```diff
@@ -857,40 +857,38 @@ def _get_metadata_from_file(meta_data):
...
+
+    # If we have no title, try to get it from document
+    if 'title' not in meta:
+        for j, line in enumerate(meta_data[i:]):
             if re_rst_title.findall(line) and i > 0:
                 meta['title'] = meta_data[i - 1].strip()
-        if 'title' not in meta:
```
Comment (Kwpolska) [inline nikola/post.py]: `not`?

Comment (Kwpolska) [inline nikola/post.py]: wait, that’s deleted, I’m blind

Comment (ralsina) [inline nikola/post.py]: still WIP, forgot to mark it, ignore for a couple of hours, sorry :-)

**Pull request merged**